Makefile

```diff
@@ -1,12 +1,10 @@
-GO=$(shell which go)
+GO ?= go
```

Comment:
IIRC there was a problem by just calling `go`: the one from the outer build (set by the github action) might be different than the one from the inner build (the one called by the builder to compile the distribution). This is typically the case when using a different go version than the default one. Using `which go` will get the path for the Go at the outer build and use that in the inner build.

Reply:
I’ve been scratching my head, and I’m pretty sure this is a non-issue. There’s nothing in the Makefile that modifies `PATH`; running `go` from within `build.sh` will run the same binary that `command -v` would show in the Makefile.

But I’ve updated the `echo`s in `build.sh` to show (e.g.) `echo "Using Go: $(command -v "$GO")"` so the log output does include the fully-resolved path. (`which` is [less portable](https://hynek.me/til/which-not-posix/).)

I did notice, however, that `GO` doesn’t apply to the `goreleaser-verify` target. It looks like the only way to override which toolchain goreleaser uses is [in the YAML](https://github.com/goreleaser/goreleaser/blob/main/internal/builders/golang/build.go#L38-L40).